docs/configuration.md

```diff
@@ -0,0 +1,34 @@
+# Nestor configuration
+
+## Context
+
+Nestor configuration contains configuration for how apps are configured and deployed.
+
+## How to use it
+
+Nestor configuration's should be in a dedicated repository, so that you can keep track of changes.
+The repository should have one branch for each step of the workflow defined in the `project.yaml`.
+
+## Details
+
+### Project
+
+The `project.yaml` contains global configuration.
+You can find default values in this file but also global configuration (e.g. docker registry, workflow's steps).
+
+### Application
+
+Applications' configurations can be found under `apps` folder.
+These configuration will be merged with the `project.yaml` configuration.
+If values are defined in both project configuration and application configuration, application will overwrite project.
```

Comment:
Maybe we can rewrite this part to explain just that application values have higher priority than the default project values?